core/src/impl/Kokkos_Profiling_C_Interface.h

```diff
@@ -63,36 +63,26 @@ struct Kokkos_Profiling_SpaceHandle {
   char name[64];
 };
 
-typedef void (*Kokkos_Profiling_initFunction)(
-    const int, const uint64_t, const uint32_t,
-    Kokkos_Profiling_KokkosPDeviceInfo*);
-typedef void (*Kokkos_Profiling_finalizeFunction)();
-typedef void (*Kokkos_Profiling_beginFunction)(const char*, const uint32_t,
-                                               uint64_t*);
-typedef void (*Kokkos_Profiling_endFunction)(uint64_t);
+using Kokkos_Profiling_initFunction = void (*)(const int, const uint64_t, const uint32_t, Kokkos_Profiling_KokkosPDeviceInfo *);
+using Kokkos_Profiling_finalizeFunction = void (*)();
+using Kokkos_Profiling_beginFunction = void (*)(const char *, const uint32_t, uint64_t *);
+using Kokkos_Profiling_endFunction = void (*)(uint64_t);
 
-typedef void (*Kokkos_Profiling_pushFunction)(const char*);
-typedef void (*Kokkos_Profiling_popFunction)();
+using Kokkos_Profiling_pushFunction = void (*)(const char *);
```

Comment:
None of these, they need to be C compatible

Reply:
oh! sorry. I'll fix that now